MetaMask/metamask-extension
# fix: UX: Multichain: Only show connection toast on the home screen

## **Description**

This PR ensures that the "Connect Account" toast only displays on the home screen (i.e. not on the send screen, swap screen, connections screen, etc)

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23733?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Go to any dapp
2. Connect Account 1
3. Switch to Account 2 in MetaMask popup
4. See the Connection toast
5. Click the Swaps button
6. Don't see the Connection toast anymore

## **Screenshots/Recordings**



### **Before**



### **After**


https://github.com/MetaMask/metamask-extension/assets/46655/eae535df-f11f-4768-bd4c-7f854ac22973



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
